src/renderers/shared/shared/__tests__/refs-test.js

```diff
@@ -392,3 +391,91 @@ describe('string refs between fiber and stack', () => {
     }
   });
 });
+
+describe('root level refs', () => {
+  it('attaches and detaches root refs in stack', () => {
+    assertForRenderer('ReactDOM');
+  });
+
+  it('attaches and detaches root refs in fiber', () => {
+    assertForRenderer('ReactDOMFiber');
```

Comment:
Generally we try to avoid code that is as dynamic as this. It makes it harder to trace what's going on.